### ISSUE TITLE
 error: generic arguments must come before the first constraint

### ISSUE THREAD

**Issue by DaviRain-Su** [paritytech/substrate#12304]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

use `pallet-uniques v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28`

```
     Compiling pallet-uniques v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28#34a06217)
     Compiling pallet-transaction-payment-rpc-runtime-api v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28#34a06217)
     Compiling pallet-chainbridge-transfer v0.1.0 (https://github.com/octopus-network/octopus-pallets.git?branch=feature/add-chainbridge-v1#6a2a29cd)
     Compiling pallet-beefy v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28#34a06217)
     Compiling pallet-octopus-lpos v4.0.0-pre.0 (https://github.com/octopus-network/octopus-pallets.git?branch=feature/add-chainbridge-v1#6a2a29cd)
     Compiling pallet-grandpa v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28#34a06217)
     Compiling pallet-babe v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28#34a06217)
     Compiling pallet-beefy-mmr v4.0.0-dev (https://github.com/paritytech/substrate.git?branch=polkadot-v0.9.28#34a06217)
  error: generic arguments must come before the first constraint
     --> /Users/davirain/.cargo/git/checkouts/substrate-7e08433d4c370a21/34a0621/frame/uniques/src/lib.rs:116:4
      |
  115 |             Success = Self::AccountId,
      |             ------------------------- constraint
  116 |             Self::Origin,
      |             ^^^^^^^^^^^^
  117 |             Self::CollectionId,
      |             ^^^^^^^^^^^^^^^^^^ generic arguments
      |
  help: move the constraint after the generic arguments
      |
  114 |         type CreateOrigin: EnsureOriginWithArg<Self::Origin, Self::CollectionId, Success = Self::AccountId>;
```

**Comment by bkchr** [comment]
It was fixed in master: https://github.com/paritytech/substrate/commit/c24431eb6a95197550b30715a4c124be1aea79de

**Comment by Roznovjak** [comment]
Would it be possible to backport the fix to 0.9.28?
